index.js

```diff
@@ -1,35 +1,23 @@
 // Public intefrace
 module.exports = promisedPipe
 
+const ensureFunctionType = (fn, i) => {
+    if (typeof fn !== 'function') {
+        throw Error(`pipe requires each argument to be a function. Argument #${i+1} is of type "${typeof fn}"`)
+    }
+}
+
+const chain = (q, fn) => q.then(fn)
```

Comment:
`chain` allow to reduce an array of function into a `.then` chain where `q` is the accumulator